lib/msal-common/src/client/SilentFlowClient.ts

```diff
@@ -40,8 +42,34 @@ export class SilentFlowClient extends BaseClient {
         request: CommonSilentFlowRequest
     ): Promise<AuthenticationResult> {
         try {
-            return await this.acquireCachedToken(request);
+            const cachedToken = await this.acquireCachedToken(request);
+
+            // if the token is not expired but must be refreshed; get a new one in the background
+            if (
+                this.lastCacheOutcome ===
+                CacheOutcome.REFRESH_CACHED_ACCESS_TOKEN
+            ) {
+                this.logger.info(
+                    "SilentFlowClient:acquireCachedToken - Cached access token's refreshOn property has been exceeded'. It's not expired, but must be refreshed."
+                );
+
+                // refresh the access token in the background
+                const refreshTokenClient = new RefreshTokenClient(
+                    this.config,
+                    this.performanceClient
+                );
+                refreshTokenClient.acquireTokenByRefreshToken(request);
```

Comment:
Please mark this appropriately, we have added a linter rule to mark floating promises.

Reply:
yea, I saw that... Since this is supposed to run in the background while the cached token is returned, can't I just add an ignore statement right above it? We're not taking any action upon the success or failure of this code.